.lgtm.yml

```diff
@@ -1,4 +1,20 @@
 extraction:
   python:
     python_setup:
-      version: 3
+      version: "3"
+  cpp:
+    prepare:
+      packages:
+      - bsdtar
+      - python3-gi
+      - libcogl-pango-dev
+      - python3-pil
+      - python3-cairo
+    after_prepare:
+    - "wget -O libxmlb.zip https://github.com/hughsie/libxmlb/archive/master.zip"
```

Comment:
Ick.  Why can't we use the meson subproject for this?

Reply:
From [the forum discussion](https://discuss.lgtm.com/t/fwupd-project-not-detected-as-c-project/1742/4?u=sam):

> Unfortunately we have a firewall in place that prevents POST requests from LGTM.com’s build servers, and git HTTP cloning use POST requests.

Reply:
How about requests using git protocol?  also filtered?

Reply:
I believe so yes, we pretty much only allow HTTP(S) traffic, and only stuff that can be cached + doesn't "effect" the outside world. There are reasons for this that I can go into if you're curious.

Reply:
Hmmm, even though the URL is https (IE: https://github.com/hughsie/fwupd/blob/master/subprojects/libxmlb.wrap) it's started as an HTTP post?

Reply:
Yup: https://git-scm.com/book/en/v2/Git-Internals-Transfer-Protocols#_http_s

Reply:
You may be able to convince it to use the "dumb http" protocol, but that would slow down all other builds that need to clone... and additionally, it would allow different artefacts / files of the repo to be cached independently, and could result in an inconsistent tree when you try to build.

Reply:
I think it's probably the best we can do given the POST constraints. Hopefully libxmlb gets dragged into Ubuntu for the next gnome release, and then we can nuke the wget.

Reply:
Well I still think that depending on a "master" snapshot is introducing some uncertainty around the build in case of an API migration or other incompatibility.  At least it should be done using a tagged tarball.

FWIW libxmlb is in the Ubuntu disco release now: https://launchpad.net/ubuntu/+source/libxmlb